### PR TITLE
feat(release): Adds btree index on `Release` version

### PR DESCRIPTION
**Pull request by ahmedetefy**
Adds a btree index on field `version` of model
`Release` to be able to perform prefix search on
release names

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
We should be able to add this to the model definition now.

In `Release.Meta` add
```
indexes = [Index(fields=["organization_id", "version"], opclasses=['', 'text_pattern_ops'])]
```

This should be roughly what we need to do here, then you should be able to generate the migration from that. Now that checked migrations are in, this should also automatically create the index using `concurrently`, so we likely don't need the custom sql anymore.

**Comment by ahmedetefy** [comment]
> We should be able to add this to the model definition now.
> 
> In `Release.Meta` add
> 
> ```
> indexes = [Index(fields=["organization_id", "version"], opclasses=['', 'text_pattern_ops'])]
> ```
> 
> This should be roughly what we need to do here, then you should be able to generate the migration from that. Now that checked migrations are in, this should also automatically create the index using `concurrently`, so we likely don't need the custom sql anymore.

@wedamija 
Would this create a btree index tho?

**Comment by wedamija** [comment]
> > We should be able to add this to the model definition now.
> > In `Release.Meta` add
> > ```
> > indexes = [Index(fields=["organization_id", "version"], opclasses=['', 'text_pattern_ops'])]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This should be roughly what we need to do here, then you should be able to generate the migration from that. Now that checked migrations are in, this should also automatically create the index using `concurrently`, so we likely don't need the custom sql anymore.
> 
> @wedamija Would this create a btree index tho?

Yeah, btree indexes are the default: https://docs.djangoproject.com/en/4.0/ref/models/indexes/#index-options

**Comment by untitaker** [comment]
I'm out of my depth here, that's an ops topic

File: src/sentry/migrations/0267_sentry_release_version_btree_idx.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    dependencies = [
+        ("sentry", "0266_add_dashboard_widget_detail_field"),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name="release",
+            index=models.Index(
+                fields=["organization", "version"],
+                name="version_btree_idx",
```
Comment (wedamija) [inline src/sentry/migrations/0267_sentry_release_version_btree_idx.py:35]: Let's call this `sentry_release_version_btree_idx`

**Review by wedamija — APPROVED** [review]

File: src/sentry/migrations/0267_sentry_release_version_btree_idx.py (hunk 1 above)
Comment (ahmedetefy) [reply src/sentry/migrations/0267_sentry_release_version_btree_idx.py]: Actually I remember why I wasn't able to do this 
`['Index names cannot be longer than 30 characters.']`

File: src/sentry/models/release.py [hunk 2]
```diff
@@ -465,6 +465,13 @@ class Meta:
             ("organization", "date_added"),
             ("organization", "status"),
         )
+        indexes = [
+            models.Index(
+                fields=["organization", "version"],
+                opclasses=["", "text_pattern_ops"],
+                name="version_btree_idx",
```
Comment (beezz) [inline src/sentry/models/release.py:472]:
```suggestion
                name="sentry_release_version_btree_idx",
```

**Review by beezz — CHANGES_REQUESTED** [review]

File: src/sentry/models/release.py (hunk 2 above)
Comment (ahmedetefy) [reply src/sentry/models/release.py:472]: @beez I renamed it to `sentry_release_version_btree` because there is a 30 char limit

Comment (beezz) [reply src/sentry/models/release.py:472]: I see you had an issue with the length of the index name but if the migration and model index names won't match on change or removal the generated SQL will again use the model index name.

Comment (ahmedetefy) [reply src/sentry/models/release.py]: Ah right, this should be fixed now!
Do I need to merge this before we run the migration or after?

Comment (wedamija) [reply src/sentry/models/release.py]: You can merge now, indexes are safe here and won't cause problems if they don't exist.

**Review by beezz — APPROVED** [review]

**Pull request merged**